File: steering_command.cpp

#include "ros/ros.h"
#include "std_msgs/Float32.h"

const int num_commands = 401;

float* CommandList(){
	static float angleList[num_commands];
	int num = (num_commands-1)/4;
	for (int i=0; i<num; i++){
		angleList[i] = -2*i;
	}
	for (int i=0; i<num; i++){
		angleList[num+i] = -200 + 2*i;
	}
	for (int i=0; i<num; i++){
		angleList[2*num+i] = 2*i;
	}
	for (int i=0; i<=num; i++){
		angleList[3*num+i] = 200 - 2*i;
	}
	return angleList;
}

int main(int argc, char **argv)
{
	ros::init(argc, argv, "steer_command");
	ros::NodeHandle n;
	ros::Publisher command_pub = n.advertise<std_msgs::Float32>("ros_talon/steering_angle", 100);
	ros::Rate loop_rate(5);
	float* angleList = CommandList(); 
	int count = 0;
	while (ros::ok() && count < num_commands)
	{
		std_msgs::Float32 angle_des;
		float angle = angleList[count];
		angle_des.data = angle;
		ROS_INFO("%d : %f", count, angle_des.data);
		command_pub.publish(angle_des);
		ros::spinOnce();
		loop_rate.sleep();
		++count;
	}
	return 0;

}

File: steering_command.py
#!/usr/bin/env python
# license removed for brevity
import rospy
from std_msgs.msg import Float32

def steer_command(command_list):
	pub = rospy.Publisher('ros_talon/steering_angle', Float32, queue_size=10)
	rospy.init_node('steer_command', anonymous=True)
	rate = rospy.Rate(1) # 10hz
	timeIndex = 0
	while not rospy.is_shutdown():
		angle_command = command_list[timeIndex]
		timeIndex += 1
		rospy.loginfo(angle_command)
		pub.publish(angle_command)
		rate.sleep()

if __name__ == '__main__':
	try:
		command_list = [0,50,100,50,0,-50,-100,-50,0]
		steer_command(command_list)
	except rospy.ROSInterruptException:
		pass